stdlib/src/collections/list.mojo

```diff
@@ -191,15 +241,28 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
 
     @always_inline
     fn _realloc(inout self, new_capacity: Int):
-        var new_data = UnsafePointer[T].alloc(new_capacity)
+        var new_data: UnsafePointer[T]
+        if new_capacity <= Self._small_buffer_size:
+            new_data = self._small_buffer.unsafe_ptr()
+        else:
+            new_data = UnsafePointer[T].alloc(new_capacity)
 
-        for i in range(self.size):
-            move_pointee(src=self.data + i, dst=new_data + i)
+        @parameter
+        if Self.sbo_enabled:
+            if self.data != new_data:
```

Comment:
**Suggestion** Include a comment here for readability, mentioning that we don't need to move anything if the data is stored in the inline buffer and the new capacity still fits in that inline buffer.